### PR TITLE
(feat, typescript): support `hasNextPage`  property for offset pagination

### PR DESCRIPTION
**Pull request by dsinghvi**
This PR upgrades the TypeScript IR by a minor version and adds support for looking at a `hasNextPage` property for pagination.

### CODE REVIEW

File: seed/ts-sdk/pagination/src/api/resources/users/client/Client.ts [hunk 1]
```diff
@@ -532,7 +532,7 @@ export class Users {
         let _offset = request?.page != null ? request?.page : 1;
         return new core.Pageable<SeedPagination.ListUsersPaginationResponse, SeedPagination.User>({
             response: await list(request),
-            hasNextPage: (response) => (response?.data ?? []).length > 0,
+            hasNextPage: (response) => response?.hasNextPage ?? (response?.data ?? []).length > 0,
```
Comment (dsinghvi) [inline seed/ts-sdk/pagination/src/api/resources/users/client/Client.ts:535]: core change

Comment (amckinney) [reply seed/ts-sdk/pagination/src/api/resources/users/client/Client.ts:535]: Sweet, deferring to the length check is great 👍

File: generators/typescript/sdk/generator/src/contexts/SdkContextImpl.ts [hunk 2]
```diff
@@ -322,4 +322,21 @@ export class SdkContextImpl implements SdkContext {
             timeoutSdkErrorGenerator
         });
     }
+
+    public isRequired(type: TypeReference): boolean {
```
Comment (amckinney) [inline generators/typescript/sdk/generator/src/contexts/SdkContextImpl.ts:326]: What about required primitive types?

Comment (amckinney) [reply generators/typescript/sdk/generator/src/contexts/SdkContextImpl.ts:326]: Should we always just `return true` after the first `if`?

File: generators/typescript/sdk/generator/src/contexts/SdkContextImpl.ts [hunk 3]
```diff
@@ -322,4 +322,21 @@ export class SdkContextImpl implements SdkContext {
...
+        if (type.type === "container" && type.container.type === "optional") {
+            return true;
```
Comment (amckinney) [inline generators/typescript/sdk/generator/src/contexts/SdkContextImpl.ts:328]: This should be `false` right?

Comment (dsinghvi) [reply generators/typescript/sdk/generator/src/contexts/SdkContextImpl.ts:328]: yup thats my b

**Review by amckinney — APPROVED** [review]

**Pull request merged**